// alpha-chatbot/chat.js


class ChatBot {
    constructor(name) {
        this.name = name || "ChatBot";

        // Default responses
        this.responses = {
            greeting: [
                "Hello!",
                "Hi there!",
                "Greetings!",
                "Hey! How can I help?"
            ],
            farewell: [
                "Goodbye!",
                "See you later!",
                "Bye! Take care!"
            ],
            default: [
                "Interesting!",
                "Tell me more.",
                "I see.",
                "Can you explain further?"
            ]
        };
    }

    /**
     * Return a random response from a category
     */
    randomResponse(category) {
        const options = this.responses[category] || this.responses.default;
        const index = Math.floor(Math.random() * options.length);
        return options[index];
    }

    /**
     * Greet a user
     */
    greet(userName) {
        return `${this.randomResponse('greeting')} ${userName || ''}`.trim();
    }

    /**
     * Farewell message
     */
    farewell(userName) {
        return `${this.randomResponse('farewell')} ${userName || ''}`.trim();
    }

    /**
     * Respond to a user's message
     */
    ask(message) {
        // You can extend this with AI/NLP logic later
        const lower = message.toLowerCase();
        if (lower.includes('hello') || lower.includes('hi')) {
            return this.randomResponse('greeting');
        } else if (lower.includes('bye') || lower.includes('goodbye')) {
            return this.randomResponse('farewell');
        } else {
            return this.randomResponse('default');
        }
    }

    /**
     * Add new custom responses
     */
    addResponses(category, responsesArray) {
        if (!this.responses[category]) this.responses[category] = [];
        this.responses[category] = this.responses[category].concat(responsesArray);
    }
}

// Export the chatbot
module.exports = ChatBot;
